### DISCUSSION TITLE
Does NBomber generate reports? Can these reports be available in GitHub Actions?

### DISCUSSION THREAD

**Discussion by PureKrome** [VIP-Logic/NBomber#569] [General]
👋🏻 G'day team!

I'm just playing around with the awesome NBomber product in some xUnit tests which will be auto ran with GitHub actions as part of CI/CD stuff. Lovely!

Now, is there a way to generate a report for each NBomber test that is ran which can integrate into GitHub actions?

In the docs, I'm seeing  a lot of lovely ASCII art .. which suggests this is a console app that is load testing some endpoint (or endpoints, if the scenario has multiple steps?).

But can any of this data be exposed into html .. to then get exposed into GitHub Action summary somewhere?

Cheers!

**Comment by AntyaDev** [comment]
Hi @PureKrome ,

Yes, NBomber generates HTML, TXT, MD, CSV reports and also supports realtime reporting.

**Comment by behdad088** [comment]
Hi @PureKrome 
NBomber can generate HTML reports for load tests. To configure the HTML reporter, install the NBomber.Reporter package and add code to your test project. You can upload the HTML report as an artifact in GitHub actions using the actions/upload-artifact action.
